### DISCUSSION TITLE
test colission

### DISCUSSION THREAD

**Discussion by jbicalhk** [fallahn/tmxlite#153] [Q&A]
hi, I want to know how I test the collisions that I indicated on tiledmaps?

**Comment by fallahn** [comment]
That's a very broad question - it depends on how you added your collision data to your maps. Generally:

- Create an object layer (or more) with your collision objects on it
- Load the map with the map loader and put all the collision objects in some kind of container like std::vector
- If the objects are all rectangular you can use them for [simple AABB collision](https://trederia.blogspot.com/2016/02/2d-physics-101-pong.html)
- If the objects are complex polygons, [deconstruct them into smaller convex shapes](https://www.emanueleferonato.com/2011/09/12/create-non-convex-complex-shapes-with-box2d/) and [use SAT](https://dyn4j.org/2010/01/sat/)

**Reply by ldornbusch** [reply]
Back in 2019 I created some methods to query a tile in a specific layer: https://github.com/fallahn/tmxlite/pull/49
I use it for my platformer: I have a separate Layer (which is optional invisible) and it shows all places the character can stand on or walls. You can check with the 
`tmx::TileLayer::Tile getTile(int tileX, int tileY)` function if a collision occurred by checking the new position. If the tile is empty the Tile Id is zero.
